linetools/isgm/io.py

```diff
@@ -41,3 +51,83 @@ def abssys_from_json(filename):
+    """
+    Parameters
+    ----------
+    filename : str
+      joeB VP filename
+    coord : SkyCoord
+    llist : LineList, optional
+    specfile : str, optional
+    chk_vel : bool, optional
+      Demand that the velocities of a given ion all be the same
+
+    Returns
+    -------
+    comps : list
+      list of AbsComponent objects
+    """
+    # init
+    if llist is None:
+        llist = LineList('ISM')
+    comps = []
+    # Read
+    vp_data = Table.read(filename, format='ascii')
+
+    # Subset by zsys + trans
+    lbls = []
+    for izsys, itrans in zip(vp_data['zsys'], vp_data['trans']):
+        lbls.append('{:.6f}_{:s}'.format(izsys, itrans))
+    lbls = np.array(lbls)
+    ulbls = np.unique(lbls)
+
+    # Subset by nflag; Build components
+    for lbl in ulbls:
+        mt_lines = np.where(lbls == lbl)[0]
+        if chk_vel:
+            if len(np.unique(vp_data['vel'][mt_lines])) != 1:
+                pdb.set_trace()
+        z_fit = vp_data['zsys'][mt_lines[0]] + vp_data['vel'][mt_lines[0]] * (1 + vp_data['zsys'][mt_lines[0]]) / ckms
+        # Loop on abs lines
+        alines = []
+        for idx in mt_lines:
+            zlim = [vp_data['zsys'][idx] +
+                    vp_data[vkey][idx] * (1 + vp_data['zsys'][idx]) / ckms
+                    for vkey in ['vlim1', 'vlim2']]
+            absline = AbsLine(vp_data['restwave'][idx] * u.AA, z=z_fit,
+                              zlim=zlim, linelist=llist)
+            # Add measurements [JB -- Want to capture anything else??]
+            absline.attrib['coord'] = coord
+            absline.attrib['flag_N'] = 1
+            absline.attrib['logN'] = vp_data['col'][idx]
+            absline.attrib['sig_logN'] = vp_data['sigcol'][idx]
+            absline.attrib['b'] = vp_data['bval'][idx]
+            absline.attrib['sig_b'] = vp_data['sigbval'][idx]
+            absline.attrib['z'] = z_fit
+            absline.attrib['sig_z'] = (1 + vp_data['z_comp'][idx]) * vp_data['sigvel'][idx] / ckms
+            if specfile is None:
+                absline.attrib['specfile'] = vp_data['specfile'][idx]
+            else:
+                absline.attrib['specfile'] = specfile
+            # Fill N, sig_N
+            _, _, = linear_clm(absline.attrib)
+            alines.append(absline)
+        # Component
+        stars = '*' * alines[0].ion_name.count('*')
+        abscomp = AbsComponent.from_abslines(alines, stars=stars)
+        # Add measurements [JB -- Want to capture anything else??]
+        abscomp.attrib = alines[0].attrib.copy()
+        # Remove undesired keys
+        for key in ['EW', 'sig_EW', 'flag_EW', 'N', 'sig_N']:
+            abscomp.attrib.pop(key)
+        # And more
+        for key in ['flag_N', 'logN', 'sig_logN']:
+            setattr(abscomp, key, abscomp.attrib[key])
+        # Errors must be in first line!
+        assert abscomp.sig_logN > 0.
+        comps.append(abscomp)
```

Comment:
"reliability" and "comment" columns should also be included.

Reply:
I'm not sure how to do this.  Going to leave for @ntejos